Update dependencies to more recent releases

lydia-duncan opened the pull request:
[reviewed by @jabraham17]

Updates:
- Sphinx from 7.2.6 to 8.1.3
- docutils from 0.20.1 to 0.21.2
- sphinx_rtd_theme from 2.0.0 to 3.0.2
- flake8 from 7.0.0 to 7.3.0
- pytest from 7.4.4 to 8.4.1
- snowballstemmer from 2.2.0 to 3.0.1

All of these are the most up-to-date version, except for Sphinx which would drop support for Python 3.10 as well as 3.9.

lydia-duncan committed:
Update the Sphinx and docutils version to the latest release

Updates:
- Sphinx from 7.2.6 to 8.0.2
- docutils from 0.20.1 to 0.21.2

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Update sphinx_rtd_theme

The old version of it required Sphinx < 8, so use the newest version now that
it is available (3.0.0)

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Resolve deprecation warning from sphinx_rtd_theme update

When compiling the documentation, we received the warning that
`get_html_theme_path` is deprecated and if it was used to define
`html_theme_path` (it was) then it can be safely removed.  So removed the lines

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Update the testing requirements as well

Updates:
- flake8 from 7.0.0 to 7.1.1
- pytest from 7.4.4 to 8.3.3

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan commented:
This looks like it will work, though the newer version of Sphinx requires Python 3.10 and up, so we'll need to double check our testing set up is ready for that before merging

mppf commented:
Currently (in prereqs.rst) we require Python 3.9 for the docs build. 3.9 in approaching end-of-life and I think it'd be completely reasonable to bump to requiring 3.10 (or even 3.11). There are plenty of not-end-of-life OSes that only have 3.9 *but* I don't think we feel obligated to make the docs build (or `start_test`) work everywhere.

In testing systems we are using Python 3.10 installed via Spack.

lydia-duncan committed:
Merge branch 'main' into updateDeps

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Update sphinx_rtd_theme again

Now that our testing machines are on later versions of Python, return to this
branch and update for more recent versions where possible

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Update snowballstemmer to the latest version

Updates from 2.2.0 to 3.0.1

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Update flake8 to the latest version

Updates from 7.1.1 to 7.3.0

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Update pytest version to the latest available

Updates from 8.3.3 to 8.4.1

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>

lydia-duncan committed:
Update Sphinx version further (but not to the most recent)

We wanted to keep support for Python 3.10, so don't update to the most recent
version available, update to 8.1.3

----
Signed-off-by: Lydia Duncan <lydia-duncan@users.noreply.github.com>
